pkg/rbd/rbd_journal.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2019 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"fmt"
+
+	"github.com/ceph/ceph-csi/pkg/util"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+)
+
+func validateNonEmptyField(field, fieldName, structName string) error {
+	if field == "" {
+		return fmt.Errorf("value '%s' in '%s' structure cannot be empty", fieldName, structName)
+	}
+
+	return nil
+}
+
+func validateRbdSnap(rbdSnap *rbdSnapshot) error {
+	if err := validateNonEmptyField(rbdSnap.RequestName, "RequestName", "rbdSnapshot"); err != nil {
```

Comment:
you can also use reflect package to get the Field name and struct name

Reply:
The one pattern I could come up with was this https://play.golang.org/p/jzhpeSPs1OW do you want this in the code, or should we let this pass till we really need reflection to be used here?

Changing the field name from a string to a string derived from the type reflection of the structure, still needs the string version to fetch the field IMO, so does not help much afaict.

If at all, the real reflection use can be to make a common validateRbdStruct, that can process either of Vol/Snap based on detecting the structure and operating on the required fields. But, I was not able to achieve this and found it easier to stick to what is above.